pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -912,8 +915,8 @@ func (iqr *IQR) AppendStatsResults(bucketHolderArr []*structs.BucketHolder, meas
 		return err
 	}
 
-	iqr.groupbyColumns = append(iqr.groupbyColumns, aggGroupByCols...)
-	iqr.measureColumns = append(iqr.measureColumns, measureFuncs...)
+	iqr.groupbyColumns = aggGroupByCols
+	iqr.measureColumns = measureFuncs
```

Comment:
It looks like you're changing this function to replace the stats results instead of appending them. In that case, you should probably rename the function

Reply:
addressed in the latest commit.